pkg/kv/kvserver/replica_application_result.go

```diff
@@ -225,6 +280,160 @@ func (r *Replica) prepareLocalResult(ctx context.Context, cmd *replicatedCmd) {
 	}
 }
 
+func (r *Replica) tryReproposeWithNewLeaseIndexV2(
+	ctx context.Context, origCmd *replicatedCmd,
+) error {
+	if origCmd.proposal.v2Reproposed {
+		// NB: we can also check v2SeenDuringApplication instead if we ever want to
+		// remove v2Reproposed. We should never hit this case: to hit it, we'd need
+		// multiple entries (i.e. copies of the same entry) of the same proposal
+		// coming up for application, each bound to a proposal. But since under V2
+		// the first entry that comes up takes the proposal out of the map (and it
+		// never re-enters the map after), the second copy cannot have reached this
+		// code path (since getting here means being associated to a ProposalData
+		// that was taken from the map).
+		return errors.AssertionFailedf("saw previously reproposed local command: %+v", origCmd.proposal)
+	}
+
+	// NB: `origCmd` remains "Local". It's just not going to signal anyone
+	// or release any latches.
+
+	origP := origCmd.proposal
+	if origP.v2Reproposed {
+		// In V2, a proposal is applied only once (see invariant in [1]). Since it
+		// has to be local to enter this branch, there is a bug - we should not see
+		// it as local twice in a row.
+		//
+		// TODO(tbg): it's likely that the `v2Reproposed` field isn't adding any
```

Comment:
I was wondering the same thing. Can we put this TODO on the field itself?

Reply:
I removed the field.